app/components-react/windows/go-live/dual-output/NonUltraDestinationSwitchers.tsx

```diff
@@ -0,0 +1,178 @@
+  (p, ref) => {
+    React.useImperativeHandle(ref, () => {
+      return {
+        addClass() {
+          addClass();
+        },
+      };
+    });
+    const containerRef = useRef<HTMLDivElement>(null);
+    const platform = typeof p.destination === 'string' ? (p.destination as TPlatform) : null;
+
+    function addClass() {
+      containerRef.current?.classList.remove(styles.platformDisabled);
+    }
+
+    function removeClass() {
+      if (p.isPrimary) {
+        alertAsync(
+          $t(
+            'You cannot disable the platform you used to sign in to Streamlabs Desktop. Please sign in with a different platform to disable streaming to this destination.',
+          ),
+        );
+        return;
+      }
+      p.onChange(false);
+      containerRef.current?.classList.add(styles.platformDisabled);
+    }
+
+    const { title, description, CloseIcon, Logo } = (() => {
+      if (platform) {
+        const { UserService } = Services;
+        // define slots for a platform switcher
+        const service = getPlatformService(platform);
+        const platformAuthData = UserService.state.auth?.platforms[platform];
+        assertIsDefined(platformAuthData);
+
+        return {
+          title: service.displayName,
+          description: platformAuthData.username,
+          Logo: () => (
+            <PlatformLogo platform={platform} className={styles[`platform-logo-${platform}`]} />
+          ),
+          CloseIcon: () => <i className={cx('icon-close', styles.close)} onClick={removeClass} />,
+        };
+      } else {
+        // define slots for a custom destination switcher
+        const destination = p.destination as ICustomStreamDestination;
+        return {
+          title: destination.name,
+          description: destination.url,
+          Logo: () => <i className={cx(styles.destinationLogo, 'fa fa-globe')} />,
+          CloseIcon: () => <i className={cx('icon-close', styles.close)} onClick={removeClass} />,
+        };
+      }
+    })();
+    return (
+      <div className={styles.platformSwitcher}>
+        <div
+          ref={containerRef}
+          className={cx(styles.switcherHeader, {
+            [styles.platformDisabled]: !p.enabled,
+          })}
+        >
+          <div className={styles.platformInfoWrapper}>
+            {/* LOGO */}
+            <Logo />
+            {/* INFO */}
+            <div className={styles.platformInfo}>
+              <span className={styles.platformName}>{title}</span>
+              <span className={styles.platformUsername}>{description}</span>
+            </div>
+            {/* CLOSE */}
+            {!p.isPrimary && <CloseIcon />}
+          </div>
+        </div>
+        <div className={styles.platformDisplay}>
+          <span className={styles.label}>{`${$t('Output')}:`}</span>
+          <DisplaySelector id={title} isPlatform={!!platform} nolabel nomargin />
+        </div>
+      </div>
+    );
+  },
+);
```

Comment:
i'm a little confused about why we need two whole components depending on user permissions here. not a blocker but feels like ripe fruit for a refactor

Reply:
While they look the same on the outside, they actually work pretty differently, solely because of the platform selector for non-ultra users. Dming more details